app/parse.py

```diff
@@ -13,10 +28,102 @@ class Product:
     price: float
     rating: int
     num_of_reviews: int
+    additional_info: dict
+
+
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+_driver: WebDriver | None = None
+
+
+def get_driver() -> WebDriver:
+    return _driver
+
+
+def set_driver(new_driver: WebDriver) -> None:
+    global _driver
+    _driver = new_driver
+
+
+def parse_additional_info(product_soup: Tag) -> dict[str, float]:
+    detailed_url = urljoin(BASE_URL, product_soup.select_one(".title")["href"])
+    driver = get_driver()
+    driver.get(detailed_url)
+    try:
+        swatches = driver.find_element(By.CLASS_NAME, "swatches")
+        buttons = swatches.find_elements(By.TAG_NAME, "button")
+        add_info = {}
+        for button in buttons:
+            if not button.get_property("disabled"):
+                button.click()
+                add_info[
+                    button.get_property("value")
+                ] = float(driver.find_element(
+                    By.CLASS_NAME, "price"
+                ).text.replace("$", ""))
+    except NoSuchElementException:
+        colors = driver.find_elements(By.TAG_NAME, "option")
+        add_info = [color.get_property("value") for color in colors][1:]
+    return add_info
+
+
+def parse_single_product(product_soup: Tag) -> Product:
+    additional_info = parse_additional_info(product_soup)
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(".description").text,
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=len(product_soup.select(".glyphicon-star")),
+        num_of_reviews=int(
+            product_soup.select_one(".ratings > p.pull-right").text.split()[0]
+        ),
+        additional_info=additional_info,
+    )
+
+
+def accept_cookies(driver: WebDriver) -> None:
+    try:
+        accept = driver.find_element(By.CLASS_NAME, "acceptCookies")
+        accept.click()
+    except NoSuchElementException:
+        pass
+
+
+def load_page(driver: WebDriver) -> None:
+    driver.maximize_window()
+    try:
+        more = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+        while more.is_displayed():
+            more.click()
+            time.sleep(1)
+    except NoSuchElementException:
+        pass
```

Comment:
DRY, you use `try/except` three times with the same error, create one function for `try/except `